cmd/soroban-cli/src/commands/tx/xdr.rs

```diff
@@ -0,0 +1,67 @@
+use std::{
+    io::{stdin, Read},
+    path::PathBuf,
+};
+
+use soroban_env_host::xdr::ReadXdr;
+use soroban_sdk::xdr::{Limits, Transaction, TransactionEnvelope};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("failed to decode XDR from base64")]
+    Base64Decode,
+    #[error("failed to decode XDR from file: {0}")]
+    FileDecode(PathBuf),
+    #[error("failed to decode XDR from stdin")]
+    StdinDecode,
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+}
+
+/// XDR input, either base64 encoded or file path and stdin if neither is provided
+#[derive(Debug, clap::Args, Clone)]
+#[group(skip)]
+pub struct Args {
+    /// Base64 encoded XDR transaction
+    #[arg(
+        long = "xdr-base64",
+        env = "STELLAR_TXN_XDR_BASE64",
+        conflicts_with = "xdr_file"
+    )]
+    pub xdr_base64: Option<String>,
+    //// File containing Binary encoded data
+    #[arg(
+        long = "xdr-file",
+        env = "STELLAR_TXN_XDR_FILE",
+        conflicts_with = "xdr_base64"
+    )]
+    pub xdr_file: Option<PathBuf>,
```

Comment:
Can we replace these xdr inputs with reading from stdin only, for consistency with the xdr commands? Would rather keep the interface smaller, and then look at improving the interface for passing XDR around separately and holistically across all the XDR commands to make sure what we do works everywhere and is consistent. The API for parameter passing is likely to be different to this to be consistent with the how the XDR subcommand does format selection.

Reply:
That's okay with me. `type` and `cat` can work for now. And then when we can properly build a txn envolope it will be the best way to start the piping.